src/epgsql_sock.erl

```diff
@@ -89,13 +91,20 @@
                 repl_feedback_required,
                 repl_cbmodule,
                 repl_cbstate,
-                repl_receiver}).
+                repl_receiver,
+                timer}).
```

Comment:
Yep, looks like single timer may be not enough. Asynchronous interfaces `epgsql{a,i}` allow request pipelining like
```
Ref1 = epgsqla:timed_equery(...),
Ref2 = epgsqla:timed_equery(...),
Ref3 = epgsqla:timed_equery(...),
Res1 = recv .. Ref1 ...,
Res2 = recv .. Ref2 ...,
Res3 = recv .. Ref3 ...,
```
Pipelining in PostgreSQL works the same way as in HTTP pipelining https://en.wikipedia.org/wiki/HTTP_pipelining

But we have a kind of dilemma there, because, AFAIK, if we have, eg, 3 pending pipelined queries and only 1st of them have timeout, if timeout on this query reached and we send `cancel`, all pending requests will generate `query_canceled`. I think this behaviour is acceptable, but should be documented.
If we have 3 pipelined requests and each of them with timeout, then we have to decide what to do: keep only 1st active timer or have 3 timers in the mode 'the one which fires earlier 1st wins'...

-------------------------------

I think for beginning we may just state that asynchronous interface with timeouts in pipelining mode have limited functionality and must be used with care and full understanding.